File: src/main/java/com/example/employeeWork/service/EmployeeServiceImpl.java
package com.example.employeeWork.service;

import com.example.employeeWork.entity.Employee;
import com.example.employeeWork.repository.EmployeeRepository;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@AllArgsConstructor
public class EmployeeServiceImpl implements EmployeeService{
    private EmployeeRepository employeeRepository;
    @Override
    public Employee createEmployee(Employee employeeWork) {
        Employee EmployeeWork = new Employee();
        return employeeRepository.save(EmployeeWork);
    }

    @Override
    public Employee getEmployeeById(Long employeeId) {
        return employeeRepository.findById(employeeId);
    }

    @Override
    public List<Employee> getAllEmployee() {
        return employeeRepository.findAll();
    }

    @Override
    public Employee updateEmployee(Employee employee) {
        Employee existingEmployee = employeeRepository.findById(employee.getId()).get();
        existingEmployee.setFirstname(employee.getFirstname());
        existingEmployee.setLastname(employee.getLastname());
        existingEmployee.setDepartment(employee.getDepartment());
        existingEmployee.setEmail(employee.getEmail());
        return null;
    }

    @Override
    public void deleteEmployee(Long employeeId) {
        employeeRepository.deleteById(employeeId);
    }
}

File: src/main/java/com/example/employeeWork/service/EmployeeServiceImpl.kt
package com.example.employeeWork.service

import com.example.employeeWork.entity.Employee
import com.example.employeeWork.repository.EmployeeRepository
import lombok.AllArgsConstructor
import org.springframework.stereotype.Service

@Service
@AllArgsConstructor
class EmployeeServiceImpl : EmployeeService {
    private val employeeRepository: EmployeeRepository? = null
    override fun createEmployee(employee: Employee): Employee {
        return employeeRepository!!.save(employee)
    }

    override fun getEmployeeById(employeeId: Long): Employee {
        val optionalEmployee = employeeRepository!!.findById(employeeId)
        return optionalEmployee.get()
    }

    override fun getAllEmployee(): MutableList<Employee?> {
        return employeeRepository!!.findAll()
    }

    override fun updateEmployee(employee: Employee): Employee {
        val existingEmployee = employeeRepository!!.findById(employee.getId()).get()
          existingEmployee.
//        existingEmployee.setLastname(employee.getLastname())
//        existingEmployee.setDepartment(employee.getDepartment())
//        existingEmployee.setEmail(employee.getEmail())
        return existingEmployee
    }

    override fun deleteEmployee(employeeId: Long) {
        employeeRepository!!.deleteById(employeeId)
    }
}
